apps/smbmount.sh

```diff
@@ -0,0 +1,288 @@
+            break
+        else
+            # Install and enable files_extrnal
+            install_and_enable_app files_external
+            # create and mount external storage to the admin group
+            MOUNT_ID=$(occ_command files_external:create "SMB$count" local null::null -c datadir="/mnt/smbshares/$count" )
+            MOUNT_ID=${MOUNT_ID//[!0-9]/}
+            occ_command files_external:applicable --add-group=admin "$MOUNT_ID" -q
+            # inform the user that mounting was successfull
+            msg_box "Your mount was successfull, congratulations!\n It is accessible in your root directory in /mnt/smbshares/$count.\nYou can now use the Nextcloud external storage app to access files there."
+            break
+        fi
+    fi
+    count=$((count+1))
+done
+return
+}
+
+mount_shares() {
+# check if any smb-share is created
+if ! grep -q /mnt/smbshares /etc/fstab
+then
+    msg_box "It seems like you have not created any SMB-Share."
+    return
+fi
+args=(whiptail --title "mount SMB-Shares" --checklist --separate-output "This option let you mount SMB-Shares to connect network-shares from the host-computer or other machines in the local network.\nChoose what you want to do.\nIf nothing is shown, then there is nothing to mount.\nSelect or unselect by pressing the spacebar" "$WT_HEIGHT" "$WT_WIDTH" 4)
+count=1
+# find out which smb-share are available
+while  [ $count -le 3 ]
+do
+    if [[ ! $(findmnt -M "/mnt/smbshares/$count") ]] && grep -q "/mnt/smbshares/$count" /etc/fstab
+    then
+        args+=("/mnt/smbshares/$count" "$(grep "/mnt/smbshares/$count" /etc/fstab | awk '{print $1}')" OFF)
+    fi
+    count=$((count+1))
+done
+# let the user choose which shares he wants to mount
+selected_options=$("${args[@]}" 3>&1 1>&2 2>&3)
+count=1
+# mount selected shares
+while  [ $count -le 3 ]
+do
+    if [[ $selected_options == *"/mnt/smbshares/$count"* ]]
+    then
+        mount "/mnt/smbshares/$count"
+        if [[ ! $(findmnt -M "/mnt/smbshares/$count") ]]
+        then
+            msg_box "It seems like the mount of /mnt/smbshares/$count wasn't successful. Please try again."
+        else
+            msg_box "Your mount was successfull, congratulations!\n It is accessible in your root directory in /mnt/smbshares/$count\nYou can now use the Nextcloud external storage app to access files there."
+        fi
+    fi
+    count=$((count+1))
+done
+return
+}
+
+show_all_mounts() {
+# if no entry created nothing to show
+if ! grep -q /mnt/smbshares /etc/fstab
+then
+    msg_box "You haven't created any SMB-Mount. So nothing to show."
+    return
+fi
+# find out which smb-shares are available
+args=(whiptail --title "list SMB-Shares" --checklist "This option let you show detailed information about your SMB-Shares.\nChoose what you want to show.\nSelect or unselect by pressing the spacebar" "$WT_HEIGHT" "$WT_WIDTH" 4)
+count=1
+while  [ $count -le 3 ]
+do
+    if grep -q "/mnt/smbshares/$count" /etc/fstab
```

Comment:
@enoch85 if I would made a variable from /mnt/smbshares, it would then be: "$SMBSHARES/$count", wouldn't it?

Reply:
this should be done